operational_analysis/toolkits/met_data_processing.py

```diff
@@ -45,26 +45,38 @@ def compute_u_v_components(wind_speed, wind_dir):
     return u, v
 
 
-def compute_air_density(df, temp_col, pres_col):
+def compute_air_density(df, temp_col, pres_col, humi_col = 'default'):
     """
-    Calculate air density from the ideal gas law given pressure and temperature.
+    Calculate air density from the ideal gas law based on the definition provided by IEC 61400-12
+    given pressure, temperature and relative humidity.
     This function assumes temperature and pressure are reported in standard units of measurement
-    (i.e. Kelvin for temperature, Pascal for pressure)
+    (i.e. Kelvin for temperature, Pascal for pressure, humidity has no dimension).
+    Humidity values are optional. According to the IEC a humiditiy of 50% (0.5) is set as default value. 
 
     Args:
         df(:obj:`pandas.DataFrame`): the input data frame with temperature and pressure data
         temp_col(:obj:`string`): column in <df> with temperature values; units of Kelvin
         pres_col(:obj:`string`): column in <df> with pressure values; units of Pascals
+        humi_col(:obj:`string`): optional column in <df> with relative humidity values; dimensionless (range 0 to 1) 
 
     Returns:
         :obj:`pandas.Series`: Rho, calcualted air density; units of kg/m3
     """
+    # Check if humidity column is provided and create default df (0.5) if necessary
+    if humi_col in df:
+        rel_humidity = df.loc[:, [humi_col]]
+    else:
+        rel_humidity = pd.DataFrame([0.5]*df[temp_col].shape[0],columns=[humi_col], index=df.index)
+        df['default'] = [0.5]*df[temp_col].shape[0]
```

Comment:
Setting this column of the input data frame may cause unintended side effects. Let's say df contains an unrelated column "default", and humi_col is set to "foo", which is not present in df. The else branch will then be taken here, and "default" will be overwritten. I suggest keeping this "default" column as a series (with a more descriptive name!) instead of adding it to the input dataframe.

Reply:
Thanks for the review. To be honest, line 70 shouldn't be existing in the pull request at all and it's not of use any in the script. I created the new data frame "rel_humidity" to avoid this issue. Line 70 can be simply deleted. The naming issue is still existing. The default value for humi_col should be probably an empty ''  I'll change that when I'm back home on Monday.